src/main/java/edu/ucsb/cs56/pconrad/springboot/hello/HelloController.java

```diff
@@ -30,5 +44,66 @@ public String profile() {
         return "profile";
     }
 
-	
+    // DEBUG
+    @RequestMapping("/test")
+    public String test() {
+        /* DEBUG
+        // DatabaseAPI.saveDataDemo();
+        // DatabaseAPI.readDataDemo();
+
+        // User a = new User("u1","u1@ucsb.edu","u001","a0a0a0");
+        // DatabaseAPI.createUser(a);
+        // User b = new User("u2","u2@ucsb.edu","u002","b0b0b0");
+        // DatabaseAPI.createUser(b);
+
+        // a = DatabaseAPI.findUser("u002");
+        // b = DatabaseAPI.findUser("u001");
+
+        // Answer a = new Answer("2018:11:30:10:18:15_u002","content01","u001");
+        // Answer b = new Answer("2018:11:30:10:18:15_u001","content02","u002");
+
+        // DatabaseAPI.composeAnswer(a);
+        // DatabaseAPI.composeAnswer(b);
+
+        // System.out.println(DatabaseAPI.retrieveUserQuestionList("u001"));
+        // System.out.println(DatabaseAPI.retrieveUserQuestionList("u002"));
+
+        System.out.println(DatabaseAPI.findAnswer("2018:11:30:11:59:24_u002"));
+
+        System.out.println(DatabaseAPI.retrieveUserAnswerList("u001"));
+
+        System.out.println(DatabaseAPI.retrieveQuestionAnswerList("2018:11:30:10:18:15_u001"));
+        */
+        System.out.println(DatabaseAPI.retrieveQuestionList());
+
+        return "test";
+    }
+
+    // DEBUG
+    @PostMapping()
+
+    // DEBUG
+    @RequestMapping(value="/testuid={uid}", method = RequestMethod.GET)
+    public String testUserProfile(@PathVariable("uid") String uid, Model model) {
+        User user = DatabaseAPI.findUser(uid);
```

Comment:
In this page, if the user does not exist, user is null which causes a NullPointerException for the below function calls.

Reply:
Resolved.

Reply:
The error still exists for instances where the value is blank, i.e something like "http://localhost:8080/testuid=".